packages/strapi-upload-aws-s3/lib/index.js

```diff
@@ -71,9 +71,10 @@ module.exports = {
         return new Promise((resolve, reject) => {
           // upload file on S3 bucket
           S3.upload({
-            Key: `${file.hash}${file.ext}`,
+            Key: `${file.path}/${file.hash}${file.ext}`,
```

Comment:
Because `file.path` is undefined, this creates an `undefined` folder in the S3 bucket an puts the file inside it.